protocol/contracts/templegold/TempleGoldVesting.sol

```diff
@@ -0,0 +1,444 @@
+        uint256 _length = ids.length;
+        summary = new VestingSummary[](_length);
+        VestingSchedule memory _schedule;
+        for (uint256 i; i < _length; ++i) {
+            _schedule = schedules[ids[i]];
+            if (_schedule.start >= _from && _schedule.start + _schedule.duration <= _to) {
+                summary[i] = VestingSummary(
+                    _schedule.recipient,
+                    _schedule.distributed,
+                    uint128(_calculateTotalVestedAt(_schedule, uint32(block.timestamp))),
+                    uint128(_calculateTotalVestedAt(_schedule, _to))
+                );
+            }
+        }
+    }
+
+    /**
+     * @notice Get vesting Ids
+     * @return ids Vesting Ids 
+     */
+    function getVestingIds() external view override returns (bytes32[] memory ids) {
+        ids = _vestingIds.values();
+    }
+    
+    /**
+     * @notice Get active vesting Ids
+     * @return ids Vesting Ids 
+     */
+    function getActiveVestingIds() external view override returns (bytes32[] memory ids) {
+        ids = _activeVestingIds.values();
+    }
+
+    /**
+     * @notice Check if id is acting vesting
+     * @param vestingId Vesting Id
+     * @return Bool
+     */
+    function isActiveVestingId(bytes32 vestingId) external view override returns (bool) {
+        return _activeVestingIds.contains(vestingId);
+    }
+
+    /**
+     * @notice Get recipient vesting count
+     * @param _recipient Recipient address
+     * @return Count
+     */
+    function getRecipientVestingCount(address _recipient) external view override returns (uint256) {
+        return _holdersVestingCount[_recipient];
+    }
+
+    /**
+     * @notice Get releasable amount
+     * @param _vestingId Vesting Id
+     * @return Amount
+     */
+    function getReleasableAmount(bytes32 _vestingId) external view override returns (uint256) {
+        if(!_vestingIdExists(_vestingId)) { return 0; }
+        VestingSchedule memory _schedule = schedules[_vestingId];
+        if (_schedule.revoked) {
+            uint256 _maxAmount = revokedMaxAmounts[_vestingId];
+            if (_maxAmount == 0) { return 0; }
+            return _calculateReleasableAmountRevoked(schedules[_vestingId], _maxAmount);
+        } else {
+            return _calculateReleasableAmount(schedules[_vestingId]);
+        }
+    }
+
+    /**
+     * @notice Get total amount vested for an Id at timestamp
+     * @param _vestingId Vesting Id
+     * @param _at At timestamp
+     * @return Total vested
+     */
+    function getTotalVestedAt(bytes32 _vestingId, uint32 _at) external view override returns (uint256) {
+        if(!_vestingIdExists(_vestingId)) { return 0; }
+        VestingSchedule memory _schedule = schedules[_vestingId];
+        if (block.timestamp <= _schedule.start) { return 0; }
+        return _calculateTotalVestedAt(_schedule, _at);
+    }
+
+    function _calculateReleasableAmount(
+        VestingSchedule memory _schedule
```

Comment:
Intended to be `memory` or `storage`? You should check other places too eg `_calculateTotalVestedAt`

Reply:
i will make a storage version of the function. some calling functions are passing `memory`. One function passes `storage`